unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/ControlsModule/SettingsControlController.cs

```diff
@@ -4,10 +4,49 @@ namespace DCL.SettingsPanelHUD.Controls
 {
     public abstract class SettingsControlController : ScriptableObject
     {
-        public abstract void Initialize();
+        protected SettingsData.GeneralSettings currentGeneralSettings;
+        protected SettingsData.QualitySettings currentQualitySetting;
+        protected ISettingsControlView view;
+
+        public virtual void Initialize(ISettingsControlView settingsControlView)
+        {
+            view = settingsControlView;
+
+            currentGeneralSettings = Settings.i.generalSettings;
+            currentQualitySetting = Settings.i.qualitySettings;
+
+            Settings.i.OnGeneralSettingsChanged += OnGeneralSettingsChanged;
+            Settings.i.OnQualitySettingsChanged += OnQualitySettingsChanged;
+        }
+
+        public virtual void OnDestroy()
+        {
+            Settings.i.OnGeneralSettingsChanged -= OnGeneralSettingsChanged;
+            Settings.i.OnQualitySettingsChanged -= OnQualitySettingsChanged;
+        }
 
         public abstract object GetStoredValue();
 
         public abstract void OnControlChanged(object newValue);
+
+        public virtual void ApplySettings()
```

Comment:
I have a question here, why we are relaying the settings applying here instead of just implementing each setting on the individual `Control` controllers? 

This will save an additional step and prevent this `Settings` class to be over-blown in the future, as right now will keep being monolithic. If we want to re-apply all the settings, we can just store the settings under a big controllers array and iterate + apply.

Reply:
As we have talked from slack, we are going to anotate this refactor as a second step to do in the near future. Meanwhile I will add this consideration in the ADR document. Thanks!